tests/test_utils.c

```diff
@@ -0,0 +1,61 @@
+#include "hircluster.h"
+
+#include "test_utils.h"
+#include <stdlib.h>
+#include <string.h>
+
+static int redis_version_major;
+static int redis_version_minor;
+
+/* Helper to extract Redis version information. */
+#define REDIS_VERSION_FIELD "redis_version:"
+void get_redis_version(redisClusterContext *cc) {
```

Comment:
A "get" function typically returns a value. Perhaps this should be called "load" or "fetch", etc. instead?

Another way would be to return the version in some struct form and later pass it to redis_version_less_than instead of using globals.

Reply:
Even simpler and without globals: We could pass cc to redis_version_less_than and fetch the version from redis every time, `redis_version_less_than(cc, major, minor)`. This is done all the time in the TCL tests for redis itself.

Reply:
Aha, yes I had it like that for a while, but then we have testcases with auth with username.
In those cases there was a need to know what to expect before connecting, and this fitted that need.

Reply:
OK, fine. :-)